l2geth/core/vm/evm.go

```diff
@@ -393,6 +396,12 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if !evm.CanTransfer(evm.StateDB, caller.Address(), value) {
 		return nil, common.Address{}, gas, ErrInsufficientBalance
 	}
+	if rcfg.UsingOVM {
+		// Make sure the creator address should be able to deploy.
+		if !evm.AddressWhitelisted(caller.Address()) {
+			return nil, common.Address{}, gas, ErrDeployerNotWhitelisted
```

Comment:
I think this is fine how it is, but could also do `fmt.Errorf("%w: %s", ErrDeployerNotWhitelisted, caller.Address().Hex())`

Reply:
I'm going to fiddle with the error so that it's clear the error comes from not being whitelisted.

Reply:
+1 that it would be very nice to show the address that is not whitelisted. Sometimes people have deployments fail not because their EOA isn't whitelisted, but because they are deploying a contract that itself does a CREATE and is not whitelisted

Reply:
Ok I figured this out. Fixed.